### PR TITLE
Add MinGW and MinGW-w64 builds to AppVeyor (take 2)

### PR DESCRIPTION
**Pull request by traversaro**
Based on https://github.com/dlfcn-win32/dlfcn-win32/pull/46 .

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -138,6 +138,7 @@ before_build:
   - set PATH=%PATH:C:\Program Files\Git\usr\bin;=%
   - set PATH=C:\MinGW\bin;%PATH%
   - if /I "%GENERATOR%" == "MinGW Makefiles" (set "GENERATOR_ARGS=") else (set "GENERATOR_ARGS=-A %PLATFORM%")
+  - if /I "%GENERATOR%" == "MinGW Makefiles" (set "INSTALL_TARGET=install") else (set "INSTALL_TARGET=INSTALL")
```
Comment (pali) [inline .appveyor.yml]: Are you sure that lowercase `install` does not work with MSVC?

File: .appveyor.yml [hunk 2]
```diff
@@ -134,10 +134,13 @@ matrix:
       CMAKE_PREFIX_PATH: "C:\\mingw-w64\\x86_64-7.2.0-posix-seh-rt_v5-rev1"
 
 before_build:
-  - set Path=C:\\MinGW\\bin;%Path%
+  # See https://help.appveyor.com/discussions/problems/3193-cmake-building-for-mingw-issue-with-git-shexe
+  - set PATH=%PATH:C:\Program Files (x86)\Git\bin;=%
```
Comment (pali) [inline .appveyor.yml]: Looks like this is more elegant solution as my original which renamed `sh.exe`.

Similar problem I had with building C module for Perl as on AppVeyor are 3 different perl versions and 2 different MinGW versions which Perl can detect. And I just renamed binaries...:
https://github.com/gooddata/DBD-MariaDB/commit/97d28bb3cf05c952c5784ed9ed1a33b06df262d6#diff-180360612c6b8c4ed830919bbb4dd459R36

File: .appveyor.yml [hunk 3]
```diff
@@ -28,62 +28,62 @@ environment:
...
     - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2015
       GENERATOR: "MinGW Makefiles"
-      CMAKE_PREFIX_PATH: "C:\\MinGW"
+      cc_path: 'C:\MinGW\bin'
       BUILD_SHARED_LIBS: OFF
 
     - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2015
       GENERATOR: "MinGW Makefiles"
-      CMAKE_PREFIX_PATH: "C:\\mingw-w64\\i686-5.3.0-posix-dwarf-rt_v4-rev0"
+      cc_path: "C:\mingw-w64\i686-5.3.0-posix-dwarf-rt_v4-rev0\mingw32\bin"
```
Comment (pali) [inline .appveyor.yml]: This is yaml string, so backslash in double quotes needs to be escaped... `"C:\\mingw-"`

File: .appveyor.yml (hunk 1 above)
Comment (pali) [reply .appveyor.yml]: lowercase `install` is working with MSVC, see: https://ci.appveyor.com/project/dlfcn-win32/dlfcn-win32/builds/22271048/job/p12xqd0frrr7oq0q#L159

File: .appveyor.yml (hunk 3 above)
Comment (traversaro) [reply .appveyor.yml]: This was fixed in the later commit: https://github.com/dlfcn-win32/dlfcn-win32/pull/48/commits/6fda813d16ea5f0881095b7fbf8b8d57a2813a39 .

File: .appveyor.yml (hunk 1 above)
Comment (traversaro) [reply .appveyor.yml]: Cool, great.

Comment (traversaro) [reply .appveyor.yml]: Fixed in https://github.com/dlfcn-win32/dlfcn-win32/pull/48/commits/56c6cbc1bb7d2d103bcec2251b54b4d69292b654 .

**Comment by traversaro** [comment]
@pali  If AppVeyor is happy and you approve the PR, I think I can squash it an merge.

**Pull request merged**